src/core/tree_methods.test.ts

```diff
@@ -0,0 +1,53 @@
+import * as tree from './tree_methods'
+
+
+let treeRoot = new tree.TreeNode("korzen")
+  .add(new tree.TreeNode("notka1"))
+  .add(new tree.TreeNode("notka2")
+    .add(new tree.TreeNode("notka1od2"))
+    .add(new tree.TreeNode("notka2od2")
+      .add(new tree.TreeNode("notka1od22"))
+      .add(new tree.TreeNode("notka2od22"))))
+  .add(new tree.TreeNode("notka3")
+    .add(new tree.TreeNode("notka1od3")));
+
+
+test("count tree nodes", () => {
+  let root = treeRoot.clone();
+  expect(tree.howManyTreeNodes(root)).toEqual(8);
+});
+
+test("add node", () => {
+  let root = treeRoot.clone();
+  let node = root.children.filter(child => (child.content === "notka3"))[0]
+  node.add(new tree.TreeNode("notatka2od3"));
+  expect(node.children.some(child => (child.content === "notatka2od3"))).toEqual(true);
+  expect(tree.howManyTreeNodes(root)).toEqual(9);
+});
+
+test("find node with string", () => {
+  let root = treeRoot.clone();
+  let nodesWithString: TreeNode[] = tree.whichTreeNodesContain((child) => child.content.includes("1od"), root);
+  expect(nodesWithString.every(obj => obj.content.includes("1od"))).toEqual(true);
+});
+
+describe("manipulatiing one node", () => {
```

Comment:
Nit: do we really need this to be a separate suite?

Reply:
nope if you dont want it, I'll scrap it